### DISCUSSION TITLE
Replacing `JSColor`.

### DISCUSSION THREAD

**Discussion by fabiocaccamo** [fabiocaccamo/django-colorfield#83] [Q&A]
`django-colorfield` is MIT licensed, but `JSColor` is GNU GPL v3 licensed.

It would be nice evaluate together the best js library for replacing `JSColor`, so anyone will be able to work on a PR.

These are the requirements for the new js library:
- 100% backward compatible, the transition must be totally transparent for who is already using this package
- vanilla js
- zero js dependencies
- cross-browser and mobile support
- MIT licensed (or MIT compatible)
- latest release < 1 year ago

Any suggestion is appreciated, votes are important too, please **don't suggest any library that doesn't match the requirements above.**

**Comment by fabiocaccamo** [comment]
[Coloris](https://github.com/mdbassit/Coloris)

**Comment by CharString** [comment]
By totally transparent, do you mean any breakage of Selenium integration tests of users is unacceptable?
Keeping the API stable is already a bit of an ask, but reasonable. Requiring the new implementation to produce the same DOM wouldn't be, up to a certain extend. Do we have a list of open projects that use django-colorfield?

**Reply by fabiocaccamo** [reply]
Requiring to produce the same DOM would be too much, I mean trying to keep the API more stable as possible and avoid CI build failure, but as always everything can be discussed for evaluating pro and cons.
